### PR TITLE
No more dataframes!

### PR DESCRIPTION
**Pull request by iskandr**
- Every predictor now has the same interface: the input is a dictionary mapping sequence identifiers to amino acid sequences, the result is a `EpitopeCollection`. 
- Captured some redundant/reoccurring code in an `EpitopeCollectionBuilder`. 
- Generalized MHC allele name normalization to work across species (preparing for John F's mouse data)
- Got rid of IEDB prediction methods which don't return IC50 values (consensus & comblib)

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/mhctools/6)

### CODE REVIEW

File: mhctools/alleles.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+    pygmy_marmoset="Cepy",
+    dusk_titi_monkey="Camo",
+    tufted_capuchin="Ceap",
+    golden_lion_tamarin="Lero",
+    white_faced_saki="Pipi",
+    saddle_backed_tamarin="Safu",
+    red_crested_tamarin="Sage",
+    moustached_tamarin="Samy",
+    cotton_top_tamarin="Saoe",
+    squirrel_moneky="Sasc",
```
Comment (arahuja) [inline mhctools/alleles.py]: monkey?

**Comment by arahuja** [comment]
<sup>**[mhctools/alleles.py, line 103 [r1]](https://reviewable.io:443/reviews/hammerlab/mhctools/6#-JnSTOcZpt1AWJWSdfTc)** ([raw file](https://github.com/hammerlab/mhctools/blob/d5aa1be6db3423d7dbac8574bdc3ef99f09b8270/mhctools/alleles.py#L103)):</sup>
NamedTuple maybe?

---

<sup>**[mhctools/alleles.py, line 134 [r1]](https://reviewable.io:443/reviews/hammerlab/mhctools/6#-JnSTET1wuaYEY6zcCBU)** ([raw file](https://github.com/hammerlab/mhctools/blob/d5aa1be6db3423d7dbac8574bdc3ef99f09b8270/mhctools/alleles.py#L134)):</sup>
Was there a second argument here at some point

---

<sup>**[mhctools/binding_measure.py, line 104 [r1]](https://reviewable.io:443/reviews/hammerlab/mhctools/6#-JnSURE4KNqH62R9gN8S)** ([raw file](https://github.com/hammerlab/mhctools/blob/d5aa1be6db3423d7dbac8574bdc3ef99f09b8270/mhctools/binding_measure.py#L104)):</sup>
or equal?

---

<sup>**[mhctools/file_formats.py, line 148 [r1]](https://reviewable.io:443/reviews/hammerlab/mhctools/6#-JnSWczUemxkZJnKi26v)** ([raw file](https://github.com/hammerlab/mhctools/blob/d5aa1be6db3423d7dbac8574bdc3ef99f09b8270/mhctools/file_formats.py#L148)):</sup>
compute the `3  + 3 * i` offset once in case this code ever needs to be updated?

---

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/mhctools/6)

**Comment by arahuja** [comment]
Overall, looks good

**Comment by iskandr** [comment]
<sup>**[mhctools/alleles.py, line 103 [r1]](https://reviewable.io:443/reviews/hammerlab/mhctools/6#-JnSTOcZpt1AWJWSdfTc)** ([raw file](https://github.com/hammerlab/mhctools/blob/d5aa1be6db3423d7dbac8574bdc3ef99f09b8270/mhctools/alleles.py#L103)):</sup>
Good idea, made a AlleleName class

---

<sup>**[mhctools/alleles.py, line 134 [r1]](https://reviewable.io:443/reviews/hammerlab/mhctools/6#-JnSTET1wuaYEY6zcCBU)** ([raw file](https://github.com/hammerlab/mhctools/blob/d5aa1be6db3423d7dbac8574bdc3ef99f09b8270/mhctools/alleles.py#L134)):</sup>
Nope, I just like having explicit tuples as arguments to format strings (otherwise, if by mistake original is itself a tuple, the string format fails)

---

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/mhctools/6)

**Pull request merged**